content/src/main/content/jcr_root/apps/acs-commons/components/authoring/templatedtextfield/render.jsp

```diff
@@ -0,0 +1,160 @@
+       */
+      - autofocus (Boolean)
+
+      /**
+       * The name of the validator to be applied. E.g. ``foundation.jcr.name``.
+       * See :doc:`validation </jcr_root/libs/granite/ui/components/foundation/clientlibs/foundation/js/validation/index>` in Granite UI.
+       */
+      - validation (String) multiple
+
+      /**
+       * The maximum number of characters (in Unicode code points) that the user can enter.
+       */
+      - maxlength (Long)
+###--%><%
+
+    Config cfg = cmp.getConfig();
+    ValueMap vm = (ValueMap) request.getAttribute(Field.class.getName());
+    Field field = new Field(cfg);
+
+    boolean isMixed = field.isMixed(cmp.getValue());
+
+    Tag tag = cmp.consumeTag();
+    AttrBuilder attrs = tag.getAttrs();
+    cmp.populateCommonAttrs(attrs);
+
+    // Start of attrs compatibility; please use cmp.populateCommonAttrs(attrs).
+    attrs.add("id", cfg.get("id", String.class));
+    attrs.addClass(cfg.get("class", String.class));
+    attrs.addRel(cfg.get("rel", String.class));
+    attrs.add("title", i18n.getVar(cfg.get("title", String.class)));
+    attrs.addOthers(cfg.getProperties(), "id", "class", "rel", "title", "type", "name", "value", "emptyText", "disabled", "required", "validation", "maxlength", "fieldLabel", "fieldDescription", "renderReadOnly", "ignoreData");
+    // End of attrs compatibility.
+
+    attrs.add("type", "text");
+    attrs.add("name", cfg.get("name", String.class));
+    attrs.add("placeholder", i18n.getVar(cfg.get("emptyText", String.class)));
+    attrs.addDisabled(cfg.get("disabled", false));
+    attrs.add("autocomplete", cfg.get("autocomplete", String.class));
+    attrs.addBoolean("autofocus", cfg.get("autofocus", false));
+
+    if (isMixed) {
+        attrs.addClass("foundation-field-mixed");
+        attrs.add("placeholder", i18n.get("<Mixed Entries>"));
+    } else {
+        attrs.add("value", vm.get("value", String.class));
+    }
+
+    attrs.add("maxlength", cfg.get("maxlength", Integer.class));
+
+    String fieldLabel = cfg.get("fieldLabel", String.class);
+    String fieldDesc = cfg.get("fieldDescription", String.class);
+    String labelledBy = null;
+
+    if (fieldLabel != null && fieldDesc != null) {
+        labelledBy = vm.get("labelId", String.class) + " " + vm.get("descriptionId", String.class);
+    } else if (fieldLabel != null) {
+        labelledBy = vm.get("labelId", String.class);
+    } else if (fieldDesc != null) {
+        labelledBy = vm.get("descriptionId", String.class);
+    }
+
+    if (StringUtils.isNotBlank(labelledBy)) {
+        attrs.add("aria-labelledby", labelledBy);
+    }
+
+    if (cfg.get("required", false)) {
+        attrs.add("aria-required", true);
+    }
+
+    String validation = StringUtils.join(cfg.get("validation", new String[0]), " ");
+    attrs.add("data-validation", validation);
+
+    attrs.addClass("coral-Textfield");
+    attrs.addClass("templated-textfield");
```

Comment:
Is there no way to reuse the standard text field code, attaching the new JS to it w/o copy/pasting the entire code to render a coral UI text field?  My concern is that as new dialog attributes are supported, this field wont support them.  Looks like other ACS Commons fields are doing the same, but thought I'd ask.

Reply:
I wanted to and originally did, but with 6.4+ they introduced Granite mixins that define what can and can't be overlaid. If I overlay the textfield itself then I add a warning to the 'health check' because i'm overlaying something AEM has said can't be overlaid.
See screens:
<img width="1115" alt="Screen Shot 2019-06-13 at 10 53 17 AM" src="https://user-images.githubusercontent.com/20479202/59451804-8d4baa80-8dc9-11e9-9a51-760dcf708153.png">
<img width="1119" alt="Screen Shot 2019-06-13 at 10 53 24 AM" src="https://user-images.githubusercontent.com/20479202/59451809-8fae0480-8dc9-11e9-993d-7494883e0068.png">

Documentation:
https://helpx.adobe.com/experience-manager/6-4/sites/deploying/using/sustainable-upgrades.html

Reply:
ya I dont think we want to "overlay" it - was wondering if there was any effective way to either extend it, or even just reference (via include) the JSP from it.  Not saying either is feasible, just wanted to see if it had been considered, since this is quite a bit of copy/paste for a field that authors will expect to work exactly like an OOTB text field with the added templating feature.

Reply:
I also tried that actually, but I need to target the field with a class (and i turn off autocomplete in the browser) that I wasn't able to add when just doing an include of the field, it turned into a whole thing. I ended up just going with the copy/paste approach. 
It also actually made the structure of the dialog break when I included it and it didn't count as part of the form anymore. It was a fun one.

Reply:
ok, I was kinda hoping that the normal rendering of a textfield included SOMETHING that related to the actual sling resource type of the field, so that by extending the type you would get something to key off of for the JS instrumentation.